### ISSUE TITLE
cli: list -w myanalysis

### ISSUE THREAD

**Issue by tiborsimko** [reanahub/reana-client#452]
**Current behaviour**

The end users can use the `-w` flag to specify wanted workflow in many `reana-client` commands, such as:

```console
$ reana-client run -w myanalysis
$ reana-client status -w myanalysis.42
$ reana-client logs -w myanalysis.42
$ reana-client ls -w myanalysis.42
$ reana-client download -w myanalysis.42
```

However, the `list` command does not accept argument:

```
$ reana-client list -w myanalysis
Usage: reana-client list [OPTIONS]
Try 'reana-client list --help' for help.

Error: no such option: -w
```

This brings a certain inconsistency in the client behaviour, especially if users launch several different workflow runs with different parameters, say `myanalysis.42`, `myanalysis.43`, `myanalysis.44`.  Currently, the users need to do:

```console
$ reana-client list | grep ^myanalysis
myanalysis           44            2020-12-08T09:35:15   2020-12-08T09:35:16   -                     running
myanalysis           43            2020-12-08T09:34:15   2020-12-08T09:34:16   2020-12-08T09:33:47   finished
myanalysis           42            2020-12-08T09:33:15   2020-12-08T09:33:16   2020-12-08T09:33:47   failed
```

to list all the various workflow runs.

**Expected behaviour**

The users may expect the `-w` argument working also for `list` command:

```console
$ reana-client list -w myanalysis
NAME                  RUN_NUMBER   CREATED               STARTED               ENDED                 STATUS
myanalysis           44            2020-12-08T09:35:15   2020-12-08T09:35:16   -                     running
myanalysis           43            2020-12-08T09:34:15   2020-12-08T09:34:16   2020-12-08T09:33:47   finished
myanalysis           42            2020-12-08T09:33:15   2020-12-08T09:33:16   2020-12-08T09:33:47   failed
```

In other words, `-w` option would perform workflow name filtering. In this way the users can use the flag "-w" basically *everywhere* where a workflow name argument option could makes sense.

(P.S. See also `--filter` and `--format` musings)

**Comment by tiborsimko** [comment]
In the meantime, we have already introduced:

```console
$reana-client list --filter name=myanalysis
```

which works nicely via substring matching.

So introducing `-w` would basically call the same function underneath, but using exact string matching instead...

**Comment by marcdiazsan** [comment]
@tiborsimko introducing the `-w` would also mean to delete the `--filter name` option? or allow them to coexist?
